package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -434,18 +434,29 @@ def guess_hydrogens(self,
         if min_mass > max_mass:
             raise ValueError("min_mass is higher than (or equal to) max_mass")
 
-        ag = self.u.select_atoms(select)
-        hydrogens_ag = ag[
-            np.logical_and.reduce((
-                ag.masses < max_mass,
-                ag.charges > min_charge,
-                ag.masses > min_mass,
-            ))
-        ]
+        if select:
+
+            ag = self.u.select_atoms(select)
```

Comment:
Ratther than doing a completely new code branch for each option, why not just do this selection solely in an if/else and leaving everything else the same?